modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -323,16 +326,38 @@ class QueueReader
 void rewindToStop(std::vector<Q*> &in_queues,
                   const std::size_t  this_id)
 {
-    for (auto &&qit : ade::util::indexed(in_queues))
-    {
-        auto id2 = ade::util::index(qit);
-        auto &q2 = ade::util::value(qit);
-        if (this_id == id2) continue;
+    // Will draining all queue & waiting for stop cmd for all queues except own queue
+    size_t expected_stop_count = std::count_if(in_queues.begin(), in_queues.end(), [] (const Q* ptr) {
+        return ptr != nullptr;
+    });
 
-        Cmd cmd;
-        while (q2 && !cv::util::holds_alternative<Stop>(cmd))
-            q2->pop(cmd);
+    // NB: extract 1 to skip own queue because stop cmd from own queue truggered rewind
```

Comment:
This text is hard to understand

Reply:
What do you think about: 
//NB: it require to substract  own queues id from total waiting queue count because it had got stop message before rewind was called
?

Reply:
Better now